lib/Parser/Parse.cpp

```diff
@@ -8443,6 +8474,19 @@ ParseNodePtr Parser::ParseExpr(int oplMin,
             // Parse the operand, make a new node, and look for more
             pnodeT = ParseExpr<buildAST>(opl, NULL, fAllowIn, FALSE, pNameHint, &hintLength, &hintOffset, nullptr);
 
+            // Detect nested function escapes of the pattern "o.f = function(){...}" or "o[s] = function(){...}".
```

Comment:
No, it won't. It's not actually meant to be exhaustive, just a cheap pattern-match to catch a certain common class of cases.